src/Rector/v8/v2/UseHtmlSpecialCharsDirectlyForTranslationRector.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v8\v2;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Ssch\TYPO3Rector\Helper\Typo3NodeResolver;
+use TYPO3\CMS\Frontend\Plugin\AbstractPlugin;
+use TYPO3\CMS\Lang\LanguageService;
+
+/**
+ * @see https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/8.2/Deprecation-71917-DeprecateTheArgumentHscForGetLLGetLLLAndSL.html
+ */
+final class UseHtmlSpecialCharsDirectlyForTranslationRector extends AbstractRector
+{
+    /**
+     * @var Typo3NodeResolver
+     */
+    private $typo3NodeResolver;
+
+    public function __construct(Typo3NodeResolver $typo3NodeResolver)
+    {
+        $this->typo3NodeResolver = $typo3NodeResolver;
+    }
+
+    /**
+     * @codeCoverageIgnore
+     */
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('htmlspecialchars directly to properly escape the content.', [
+            new CodeSample(<<<'PHP'
+use TYPO3\CMS\Frontend\Plugin\AbstractPlugin;
+class MyPlugin extends AbstractPlugin
+{
+    public function translate($hsc): void
+    {
+        $translation = $this->pi_getLL('label', '', true);
+        $translation2 = $this->pi_getLL('label', '', false);
+        $translation3 = $this->pi_getLL('label', '', $hsc);
+        $translation9 = $GLOBALS['LANG']->sL('foobar', true);
+        $translation10 = $GLOBALS['LANG']->sL('foobar', false);
+    }
+}
+PHP
+                , <<<'PHP'
+use TYPO3\CMS\Frontend\Plugin\AbstractPlugin;
+class MyPlugin extends AbstractPlugin
+{
+    public function translate($hsc): void
+    {
+        $translation = htmlspecialchars($this->pi_getLL('label', ''));
+        $translation2 = $this->pi_getLL('label', '');
+        $translation3 = $this->pi_getLL('label', '', $hsc);
+        $translation9 = htmlspecialchars($GLOBALS['LANG']->sL('foobar'));
+        $translation10 = $GLOBALS['LANG']->sL('foobar');
+    }
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
```

Comment:
I'm trying to finish the https://github.com/sabbelasichon/typo3-rector/pull/1651, and looking into this as you suggested.

I don't get my head around this code though.

The `rector()`-function calls `$this->shouldSkip()`. `$this->shouldSkip()` calls the `$this->isLanguageServiceCall()`. But the`rector()`-function itself also calls the `$this->isLanguageServiceCall()` and that I don't get.

What is the difference when calling it from the `$this->shouldSkip()` and the `rector()`-function ?

Reply:
The shouldSkip function does also a check if it is a call from an object of class AbstractPlugin.